### PR TITLE
JIT: Add fgRedirectTrueEdge, fgRedirectFalseEdge

### PR DESCRIPTION
**Pull request by amanasifkhalid**
Follow-up to #99362. Adds helper methods for redirecting BBJ_COND blocks' true/false targets to new successors while reusing the existing flow edge.

cc @dotnet/jit-contrib, @AndyAyersMS PTAL. This may conflict with #99541; I'm ok with waiting for that to be merged first.

### CODE REVIEW

**Comment by amanasifkhalid** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=598510&view=ms.vss-build-web.run-extensions-tab) are quite big, though the bulk of them seem to come from a few collections. I'll dig into these locally.

**Comment by amanasifkhalid** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=599019&view=ms.vss-build-web.run-extensions-tab) are more reasonable now that #99541 has been checked in. Looking at them locally, edge weights diffs are driving churn in `fgReorderBlocks`.

File: src/coreclr/jit/fgflow.cpp [hunk 1]
```diff
@@ -449,6 +449,84 @@ void Compiler::fgRedirectTargetEdge(BasicBlock* block, BasicBlock* newTarget)
     newTarget->bbRefs++;
 }
 
+void Compiler::fgRedirectTrueEdge(BasicBlock* block, BasicBlock* newTarget)
```
Comment (AndyAyersMS) [inline src/coreclr/jit/fgflow.cpp:452]: Please add a header comment block for this method

File: src/coreclr/jit/fgflow.cpp [hunk 2]
```diff
@@ -449,6 +449,84 @@ void Compiler::fgRedirectTargetEdge(BasicBlock* block, BasicBlock* newTarget)
...
+    }
+
+    newTarget->bbRefs++;
+
+    // Pred list of target should (stilL) be ordered
+    //
+    assert(newTarget->checkPredListOrder());
+}
+
+void Compiler::fgRedirectFalseEdge(BasicBlock* block, BasicBlock* newTarget)
```
Comment (AndyAyersMS) [inline src/coreclr/jit/fgflow.cpp:491]: Please add a header comment block for this method

File: src/coreclr/jit/fgflow.cpp [hunk 3]
```diff
@@ -449,6 +449,84 @@ void Compiler::fgRedirectTargetEdge(BasicBlock* block, BasicBlock* newTarget)
...
+    else
+    {
+        trueEdge->setNextPredEdge(predEdge);
+        trueEdge->setDestinationBlock(newTarget);
+        *predListPtr = trueEdge;
+    }
+
+    newTarget->bbRefs++;
+
+    // Pred list of target should (stilL) be ordered
```
Comment (AndyAyersMS) [inline src/coreclr/jit/fgflow.cpp:486]:
```suggestion
    // Pred list of target should (still) be ordered
```

File: src/coreclr/jit/fgflow.cpp [hunk 4]
```diff
@@ -449,6 +449,84 @@ void Compiler::fgRedirectTargetEdge(BasicBlock* block, BasicBlock* newTarget)
...
+    else
+    {
+        falseEdge->setNextPredEdge(predEdge);
+        falseEdge->setDestinationBlock(newTarget);
+        *predListPtr = falseEdge;
+    }
+
+    newTarget->bbRefs++;
+
+    // Pred list of target should (stilL) be ordered
```
Comment (AndyAyersMS) [inline src/coreclr/jit/fgflow.cpp:525]:
```suggestion
    // Pred list of target should (still) be ordered
```

**Review by AndyAyersMS — COMMENTED** [review]
Code LGTM. Left some notes on comments.

**Comment by amanasifkhalid** [comment]
@AndyAyersMS thanks for the review -- I incorporated your feedback.

**Review by AndyAyersMS — APPROVED** [review]

**Pull request merged**